Sketcher/OffsetInSketcher.FCMacro

```diff
@@ -0,0 +1,431 @@
+				self.StartPointId = 2
+				self.EndPointId = 1
+			self.Center = Geometry.Center
+			self.Radius = Geometry.Radius
+
+	def addGeometry(self,Mode=False):
+		self.GeoId = Sketch.getGeoNumber()
+		Sketch.addGeometry(self.Geo,Mode)
+
+	def reverse(self):
+		self.Geo.reverse()
+		self.StartPoint = self.Geo.StartPoint
+		self.EndPoint = self.Geo.EndPoint
+		self.StartPointId = 3-self.StartPointId
+		self.EndPointId = 3-self.EndPointId
+		if self.Type == 'arc':
+			self.Axis = self.Geo.Axis
+
+	def isNegative(self):
+		if self.Type == 'arc': 
+			if self.Axis.z < 0.0:
+				return True
+
+class Sketch:
+#
+# manage all connections to the sketch
+#
+	def addGeometry(Element,Mode):
+		ActiveSketch.addGeometry(Element,Mode)
+
+	def getGeometry():
+		return ActiveSketch.Geometry
+
+	def getGeoNumber():
+		return ActiveSketch.GeometryCount
+
+	def getSketchPoint(Element):
+		return ActiveSketch.getPoint(Element,1)
+
+	def coincident(Element1,Point1,Element2,Point2):
+		ActiveSketch.addConstraint(Sketcher.Constraint('Coincident',Element1.GeoId,Point1,Element2.GeoId,Point2))
+
+	def equal(Element1,Element2):
+		ActiveSketch.addConstraint(Sketcher.Constraint('Equal',Element1.GeoId,Element2.GeoId))
+
+	def perpendicular(Element1,Element2):
+		ActiveSketch.addConstraint(Sketcher.Constraint('Perpendicular',Element1.GeoId,Element2.GeoId))
+
+	def pointOnObject(Element1,Point,Element2):
+		ActiveSketch.addConstraint(Sketcher.Constraint('PointOnObject',Element1.GeoId,Point,Element2.GeoId))
+
+	def symmetric(Element,MidVertexNum):
+		ActiveSketch.addConstraint(Sketcher.Constraint('Symmetric',Element.GeoId,1,Element.GeoId,2,MidVertexNum,1))
+
+	def tangent(Element1,Point1,Element2,Point2):
+		ActiveSketch.addConstraint(Sketcher.Constraint('Tangent',Element1.GeoId,Point1,Element2.GeoId,Point2))
+
+class Tools:
+#
+# diverse geometric tools 
+#
+	def createLine(Point1,Point2):
+		return Edge(Part.LineSegment(Point1,Point2))
+
+	def concentricArcOfCircle(Element,Distance,Reverse=False):
+		Arc = Element.Geo.clone()
+		Circle = Arc.Circle
+		ConcentricEdge = Edge(Part.ArcOfCircle(Part.Circle(Circle,Distance),Arc.FirstParameter,Arc.LastParameter))
+		if Reverse: ConcentricEdge.reverse() # change sense of rotation
+		return ConcentricEdge
+
+	def getMidPoint(Element):
+		XPoint = (Element.EndPoint.x+Element.StartPoint.x)/2
+		YPoint = (Element.EndPoint.y+Element.StartPoint.y)/2
+		return App.Vector(XPoint,YPoint,0)
+
+class OffsetInSketcher():
+#
+# class containing the addOffset command
+#
+	def addOffset(SelGeoId,OffsetLine):
+	#
+	# this is the OffsetInSketcher command
+	#
+		def getSelectedEdges(SelGeoID):
+		#
+		# returns a list of all selected lines and arcs
+		#
+			AllGeo = Sketch.getGeometry() # full geometry list
+			Selected = []
+			for Id in SelGeoID:
+				Goemetry = AllGeo[Id]
+				Selected.append(Edge(Goemetry,Id))
+			return Selected
+
+		def areEqual(V1,V2):
+		#
+		# check the  identity of two vectors
+		#
+			if (V1-V2).Length < 0.001: return True
```

Comment:
Clearer: `return (V1-V2).Length < 0.001`, though the returned value is different. Your function return `None` if the vectors are not equal, which is quite unusual. My proposal returns `False`.